### PR TITLE
docs(sending-message): sending message c4 diagrams 

### PR DESCRIPTION
**Pull request by acuiuli**
#### List of Changes

- [docs(sending-message): add message-sender system diagram](https://github.com/pagopa/io-messages/commit/08e7d58ef1b1c9ce100414ce0912ce420f34a1d2)
- [docs(sending-message): add fn-services component diagram](https://github.com/pagopa/io-messages/commit/1b9deb0325e02355d62a28f9179f9167cea272a1)

#### Motivation and Context

Added two c4 diagrams to describe how a message is sent. One diagram is a container diagram of the message-sender  system and the other is a component diagram of the sending message functionality of io-fn-services.

#### How Has This Been Tested?

No tests are needed, just documentaions

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/sending-message/fn-services.plantuml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ContainerDb(msg_content, "message-content", "container: storage", "immutable storage for the message content")
+ContainerDb(cosmos, "IO-P-COSMOS-API", "container: cosmos", "Stores messages and user profile content")
+
+System_Ext(notif_sender, "Notification Sender System", "The system that send push notifications to mobile devices and emails")
+System_Ext(msg_reader, "Message Reader System", "The system used to read messages")
+
+Container_Boundary(io-fn-services, "io-fn-services") {
+    ContainerDb(msg_created, "MessageCreated", "container: queue", "The queue that stores the messages that needs to be processed")
+    ContainerDb(msg_processed, "MessageProcessed", "container: queue", "The queue that stores the processed messages")
+
```
Comment (Garma00) [inline docs/sending-message/fn-services.plantuml]: I would add the queue with failed process messages so we can highlight that the OnFailedProcessMessage is triggered by this queue.

**Comment by Garma00** [comment]
Also, we have a discusion [here](https://github.com/pagopa/io-messages/pull/18#issuecomment-2326201718) on naming convention for components on those diagrams

File: docs/sending-message/fn-services.plantuml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+@startuml fn-services
+
+!includeurl https://raw.githubusercontent.com/RicardoNiepel/C4-PlantUML/master/C4_Component.puml
```
Comment (lucacavallaro) [inline docs/sending-message/fn-services.plantuml]:
```suggestion
!include <C4/C4_Context>
```

File: docs/sending-message/fn-services.plantuml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+LAYOUT_WITH_LEGEND()
+
+title Component diagram for io-function-services
```
Comment (lucacavallaro) [inline docs/sending-message/fn-services.plantuml]: What is the actual system we are modeling? It seems that the scope is broader than `io-function-services`

File: docs/sending-message/fn-services.plantuml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+Container(apim, "IO-P-APIM-V2", "container: io-apim", "Check authentication, authz and rate limit on api call")
+Container(msg_sending, "io-fn-msgs-sending", "container: function-app", "Retrieve remote content configuration")
+ContainerDb(cosmos, "IO-P-COSMOS-API", "container: cosmos", "Stores messages and user profile content")
+ContainerDb(msg_content, "message-content", "container: storage", "immutable storage for the message content")
+ContainerDb(cosmos, "IO-P-COSMOS-API", "container: cosmos", "Stores messages and user profile content")
+
+System_Ext(notif_sender, "Notification Sender System", "The system that send push notifications to mobile devices and emails")
+System_Ext(msg_reader, "Message Reader System", "The system used to read messages")
+
+Container_Boundary(io-fn-services, "io-fn-services") {
```
Comment (lucacavallaro) [inline docs/sending-message/fn-services.plantuml:20]: Are you sure that `services` is the right boundary?

**Review by lucacavallaro — CHANGES_REQUESTED** [review]

File: docs/sending-message/fn-services.plantuml (hunk 3 above)
Comment (acuiuli) [reply docs/sending-message/fn-services.plantuml]: We are modeling the `io-function-services` container using a component diagram. There are also the interactions of the `io-function-services `container with external system/container

File: docs/sending-message/fn-services.plantuml (hunk 1 above)
Comment (acuiuli) [reply docs/sending-message/fn-services.plantuml]: this has been fixed [here](https://github.com/pagopa/io-messages/pull/20/commits/c8b2c54c0f6feac9b3092c2a61efd7484b7a4166)

**Review by Garma00 — APPROVED** [review]

**Review by lucacavallaro — APPROVED** [review]

**Pull request merged**